main.tf

```
@@ -19,24 +19,25 @@ resource "aws_ec2_transit_gateway" "default" {
 }
 
 resource "aws_ec2_transit_gateway_route_table" "default" {
-  count              = var.create_transit_gateway_route_table ? 1 : 0
+  count              = module.this.enabled && var.create_transit_gateway_route_table ? 1 : 0
   transit_gateway_id = local.transit_gateway_id
   tags               = module.this.tags
 }
 
 # Need to find out if VPC is in same account as Transit Gateway.
 # See resource "aws_ec2_transit_gateway_vpc_attachment" below.
 data "aws_ec2_transit_gateway" "this" {
-  id = local.transit_gateway_id
+  count = (module.this.enabled && ((var.existing_transit_gateway_id != null && var.existing_transit_gateway_id != "") || var.create_transit_gateway)) ? 1 : 0
```

This may be more readable like this

```hcl
locals {
  enabled = module.this.enabled

  # NOTE: This is the same logic as local.transit_gateway_id but we cannot reuse that local in the data source or
  # we get the dreaded error: "count" value depends on resource attributes
  create_or_existing_enabled = local.enabled && (var.create_transit_gateway || try(length(var.existing_transit_gateway_id), "") > 0)) ? 1 : 0
}
```
Then
```suggestion
  count = local.create_or_existing_enabled ? 1 : 0
```

---

hi sorry for the delay. I've been busy lately. I've pushed this with a slight change of name